src/extension/monitor/core/src/main/java/org/geoserver/monitor/MonitorFilter.java

```diff
@@ -231,26 +243,40 @@ String getHttpReferer(HttpServletRequest req) {
         }
     }
 
+    /**
+     * Audit consumer function to be executed on the underlying PostProcessTask thread. Will receive
+     * {@link RequestData} and {@link Authentication} from thread execution.
+     */
+    public void setExecutionAudit(BiConsumer<RequestData, Authentication> executionAudit) {
```

Comment:
Since it's a testing facility I'd make the methods around it at least package private. 
Another possible approach would be to manually register a RequestPostProcessor that does the same work using GeoServerExtensionsHelper.singleton

Reply:
Fixed, thanks.